_posts/2016-08-14-three-pillars-of-continuous-delivery.md

```diff
@@ -0,0 +1,47 @@
+
+**Use the right tools** Identify the tools which can help you. In the modern world of software development we have plenty of options. Tools like Jenkins, Bamboo, Travis, Ansible, Docker can help you in the automation, but without a good justification of usage they can be just fancy tools. Your automation should evolve together with your needs and problems which are introduced by these problems. Proofs of concept will allow you to understand if a specific tool is adding any value in the context of automation.
+
+### Rapid Response
+
+**Give answers now** Continuous deployment should help you to discover problems before they occur in the production environment. If your deployment process takes days, weeks or even hours in some cases, you can miss the opportunity to find out if your software is valid and ready to use. The time between the discovery and the fix of an issue can be minimised by the rapid response from your continuous deployment process.
+
+**Make it faster**  Many different things can make your process slower. One such problem is how your processes are organised. In many cases in my previous projects the original version of the process required the individual involvement of many people in the organisation. Sometimes you don't have the right access rights to your infrastructure and deployments are impossible. Try to improve or change these elements in your organisation. The automation of your process should not only include the solutions for technical problems, but it should also improve non-technical aspects of the process.
+
+### Team Effort
+
+**Do it together** In my opinion the responsibility for continuous delivery should be shared by the team. By the team I mean all people involved in the project. Not only software developers, not only system administrators, we should also involve business stakeholders. Understanding continuous deployment, by all members of the team, strengthens knowledge about the software and improves cooperation. Our software is no longer just lines of code, it is also the place where this software lives.
+
+**No Silo Culture** Every technical person in the team should be able to start, improve and fix the continuous delivery process. I understand that this can be a challenge and many failures can appear, but only including people make your continuous delivery useful. Without the engagement of your whole team you create an invisible wall between the software development and its deployment. This can ruin all efforts of introducing automation, because your process will not be aligned with changes taking place in the code.
```

Comment:
I understand that this can be a challenge and the team may face many failures, but by being inclusive, you can make your continuous delivery a beneficial process.